src/control/server/mgmt_svc_test.go

```diff
@@ -130,3 +138,96 @@ func TestCheckMgmtSvcReplica(t *testing.T) {
 		}
 	}
 }
+
+func newTestGetACLReq() *mgmtpb.GetACLReq {
+	return &mgmtpb.GetACLReq{
+		Uuid: "testUUID",
+	}
+}
+
+func TestPoolGetACL_NoMS(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newMgmtSvc(NewIOServerHarness(log))
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	cmpErr(t, errors.New("no managed instances"), err)
+}
+
+func TestPoolGetACL_DrpcFailed(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newTestMgmtSvc(log)
+	expectedErr := errors.New("mock error")
+	setupMockDrpcClient(svc, nil, expectedErr)
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	cmpErr(t, expectedErr, err)
+}
+
+func TestPoolGetACL_BadDrpcResp(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newTestMgmtSvc(log)
+	// dRPC call returns junk in the message body
+	badBytes := make([]byte, 12)
+	for i := range badBytes {
+		badBytes[i] = byte(i)
+	}
+
+	setupMockDrpcClientBytes(svc, badBytes, nil)
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	cmpErr(t, errors.New("unmarshal"), err)
+}
+
+func TestPoolGetACL_Success(t *testing.T) {
```

Comment:
Maybe add a test for a bad or malformed uuid.

Reply:
There is a test on the C side where the UUID checking is currently implemented. If we want to bail earlier, I think it should be at the top level, in the PoolGetACLCmd in the daos_shell tool.